test/topicFilters.spec.ts

```diff
@@ -0,0 +1,296 @@
+      "Priority_1",
+      "(priority = 1 OR priority = 2) AND (sys.label LIKE '%String2')"
+    );
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "Priority_1");
+  });
+
+  it("Add Rule with SQL filter and action", async function(): Promise<void> {
+    await subscriptionClient.addRule(
+      "Priority_1",
+      "(priority = 1 OR priority = 3) AND (sys.label LIKE '%String1')",
+      "SET sys.body = 'MessageX'"
+    );
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "Priority_1");
+  });
+
+  it("Add Rule with SQL filter and action", async function(): Promise<void> {
+    await subscriptionClient.addRule(
+      "Priority_1",
+      "(priority = 1 OR priority = 3) AND (sys.label LIKE '%String1')",
+      "SET sys.body = 'MessageX'"
+    );
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "Priority_1");
+  });
+
+  it("Add Rule with Correlation filter", async function(): Promise<void> {
+    await subscriptionClient.addRule("Correlationfilter", {
+      label: "red",
+      correlationId: "high"
+    });
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "Correlationfilter");
+  });
+
+  it("Add Default rule on a subscription that its default rule removed", async function(): Promise<
+    void
+  > {
+    await subscriptionClient.addRule("$Default", {
+      label: "red",
+      correlationId: "high"
+    });
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules[0].name, "$Default");
+  });
+
+  it("Adding a rule with a name which matches with existing rule", async function(): Promise<void> {
+    await subscriptionClient.addRule("Priority_1", "priority = 1");
+    let errorWasThrown = false;
+    try {
+      await subscriptionClient.addRule("Priority_1", "priority = 2");
+    } catch (error) {
+      errorWasThrown = true;
+      should.equal(!error.message.search(" already exists."), false);
+      console.log(error.message);
+    }
+    should.equal(errorWasThrown, true);
+  });
+
+  it("Adding a rule with no name", async function(): Promise<void> {
+    let errorWasThrown = false;
+    try {
+      await subscriptionClient.addRule("", "priority = 2");
+    } catch (error) {
+      errorWasThrown = true;
+      should.equal(!error.message.search("name is missing"), false);
+      console.log(error.message);
+    }
+    should.equal(errorWasThrown, true);
+  });
+
+  it("Adding a rule with no filter", async function(): Promise<void> {
+    let errorWasThrown = false;
+    try {
+      await subscriptionClient.addRule("Priority_1", "");
+    } catch (error) {
+      errorWasThrown = true;
+      should.equal(!error.message.search("Filter is missing"), false);
+      console.log(error.message);
+    }
+    should.equal(errorWasThrown, true);
+  });
+
+  it("Adding a rule with a Boolean filter whose input is not Boolean", async function(): Promise<
+    void
```

Comment:
Since we dont have separate function to add a "boolean" filter, this (and the commented test case after this) doesnt make sense....

How about changing this to `Adding a rule with a filter that is neither boolean, string or a co-relation object`

Reply:
Yes, already noted this. I wanted to add something like this.